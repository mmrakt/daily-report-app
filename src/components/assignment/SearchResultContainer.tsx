import React from 'react'
import { useFetchCategories } from '@/hooks/category/useFetchCategories'
import { useFetchProjects } from '@/hooks/project/useFetchProjects'
import LoadingSpinner from '../common/LoadingSpinner'
import CategoriesContainer from './CategoriesContainer'
import ProjectsContainer from './ProjectsContainer'

const SearchResultContainer: React.FC<{ roleId: number }> = ({ roleId }) => {
    const { data: categories, isLoading: isLoadingFetchCategories } =
        useFetchCategories()
    const { data: projects, isLoading: isLoadingFetchProjects } =
        useFetchProjects()

    if (isLoadingFetchCategories || isLoadingFetchProjects)
        return <LoadingSpinner />

    return (
        <>
            <CategoriesContainer categories={categories} roleId={roleId} />
            <ProjectsContainer projects={projects} roleId={roleId} />
        </>
    )
}

export default SearchResultContainer
