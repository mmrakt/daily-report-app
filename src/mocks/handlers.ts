import { rest } from 'msw'
import categories from './api/categories'
import projects from './api/projects'

export const handlers = [
    rest.get(`http://localhost:3000/api/categories`, categories.get),
    rest.get(`http://localhost:3000/api/projects`, projects.get),
]
