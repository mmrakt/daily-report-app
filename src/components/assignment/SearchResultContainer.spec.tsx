import { server } from '@/mocks/server'
import { render, waitForElementToBeRemoved } from '@testing-library/react'
import SearchResultContainer from './SearchResultContainer'
import React from 'react'
import { createQueryWrapper } from '../../../test/utlis/createQueryWrapper'

describe('SearchResultContainer', () => {
    beforeAll(() => server.listen())
    afterEach(() => server.resetHandlers())
    afterAll(() => server.close())

    test('render:loading', async () => {
        const { queryWrapper } = createQueryWrapper()
        const { asFragment, getByRole } = render(
            <SearchResultContainer roleId={1} />,
            {
                wrapper: queryWrapper,
            }
        )
        expect(asFragment()).toMatchSnapshot()
        const LoadingSpinner = getByRole('loadingSpinner')
        await waitForElementToBeRemoved(LoadingSpinner)
    })

    test('render:fetch categories and projects', async () => {
        const { queryWrapper } = createQueryWrapper()
        const { findByText, queryByText } = render(
            <SearchResultContainer roleId={1} />,
            {
                wrapper: queryWrapper,
            }
        )
        expect(await findByText('category001')).toBeInTheDocument()
        expect(await findByText('project001')).toBeInTheDocument()
        expect(await queryByText('project004')).not.toBeInTheDocument()
    })
})
