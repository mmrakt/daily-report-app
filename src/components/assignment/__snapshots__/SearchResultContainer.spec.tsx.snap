// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultContainer render:categories and projects 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center"
  >
    <div
      class="animate-spin h-10 w-10 border-4 rounded-full border-t-transparent"
    />
  </div>
</DocumentFragment>
`;

exports[`SearchResultContainer render:loading 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center"
    role="loadingSpinner"
  >
    <div
      class="animate-spin h-10 w-10 border-4 rounded-full border-t-transparent"
    />
  </div>
</DocumentFragment>
`;
